packages/ckeditor5-bookmark/src/bookmarkediting.ts

```diff
@@ -112,3 +127,25 @@ export default class BookmarkEditing extends Plugin {
 		} );
 	}
 }
+
+/**
+ * A helper function to match an `anchor` element which must contain `id` attribute but without `href` attribute,
+ * also element must be empty when matcher is execute in data pipeline so it can be upcasted to a `bookmark`.
+ */
+function upcastMatcher( element: ViewElement, dataPipeline: boolean = true ) {
+	const isAnchorElement = element.name === 'a';
+
+	if ( !isAnchorElement ) {
+		return null;
+	}
+
+	const hasIdAttribute = element.hasAttribute( 'id' );
+	const hasHrefAttribute = element.hasAttribute( 'href' );
+	const isEmpty = element.isEmpty;
+
+	if ( !hasIdAttribute || hasHrefAttribute || dataPipeline && !isEmpty ) {
```

Comment:
So if I understand it correctly in editingDowncast `dataPipeline` is set on `false` (L60) so `anchor` element could be not empty?

Reply:
It's not empty cause in editing we will put there an `UIElement` width bookmark icon.